pkgs/development/ocaml-modules/httpaf/default.nix

```diff
@@ -11,14 +11,14 @@ buildDunePackage rec {
     sha256 = "0i2r004ihj00hd97475y8nhjqjln58xx087zcjl0dfp0n7q80517";
   };
 
-  buildInputs = [ alcotest ];
+  buildInputs = lib.optional doCheck alcotest;
```

Comment:
put into checkInputs? also avoids having alcotest in the closure.

Reply:
Which closure?

Reply:
buildInputs usually are meant to be used at runtime, if a store path gets added somehow during the build, then the package closure will include it.

Reply:
I don’t think so: build inputs are primarily used at build time.

The package closure is computed from its contents: the details of the source of the derivation are irrelevant (if they don’t change the build results).